syntax = "proto3";

import "rustproto.proto";

option (rustproto.expose_oneof_all) = true;

message Event {

  string device = 1;

  enum EventType {
    SWITCH_OPEN = 0;
    SWITCH_CLOSED = 1;
  }

  EventType type = 2;
}

message LightControl {
  uint32 device = 1;
  uint32 level = 2;
}

message Ack {
  bool success = 1;
}

message CmdMsg {
  oneof cmd {
    LightControl lightControl = 1;
  }
}

service VinsteonRPC {
  rpc SendCmd(CmdMsg) returns (Ack) {}
  rpc SendCmdReliable(CmdMsg) returns (Ack) {}
}
